### PR TITLE
docs/dev: Add backport instructions for contributors

### PR DESCRIPTION
**Pull request by cvybhu**
Add instructions on how to backport a feature to on older version of Scylla.

It contains a detailed step-by-step instruction so that people unfamiliar with intricacies of Scylla's repository organization can easily get the hang of it.

This is the guide I wish I had when I had to do my first backport.

I put it in backport.md because that looks like the file responsible for this sort of information.
For a moment I thought about `CONTRIBUTING.md`, but this is a really short file with general information, so it doesn't really fit there. Maybe in the future there will be some sort of unification (see #12126)

### CODE REVIEW

**Comment by cvybhu** [comment]
At first I put it in the wiki (https://github.com/scylladb/scylladb/wiki/Backporting), but I heard that it's better to put it in `docs/dev`, so I will remove the wiki page once the PR is merged.

File: docs/dev/backport.md
Comment (denesb) [inline docs/dev/backport.md:51]: The normal flow is the following:
* PR gets merged to next
* promotion merges next to master
* at this point, if the PR fixes an issue, said issue gets the "Backport Candidate" label
* at some later point a maintainer (Avi) goes through all issues that have said label and backport it to all affected releases, if they see it as a good backport candidate
* the "Backport Candidate" label is removed from the issue

A separate backport PR is required only if the backport is non-trivial: cherry-picking it onto the target branch generates non-trivial merge conflicts. In this case the original author of said PR is asked to prepare a backport PR.

File: docs/dev/backport.md
Comment (denesb) [inline docs/dev/backport.md:63]: You might need to do another `git pull origin branch-5.0` before this, as the first fetch might fail to get all the submodule changes (as it is executed before `git submodule sync`).

File: docs/dev/backport.md
Comment (denesb) [inline docs/dev/backport.md:72]: Usually a better alternative is to cherry-pick individual commits. The backport will create a new merge commit.
This method will result in less conflicts at a time. On the other hand you might have to resolve similar conflicts multiple times.

Comment (denesb) [reply docs/dev/backport.md:72]: We should also add an example for a single-commit backport as well (basically the same, just without the `-m1` part).

**Review by denesb — CHANGES_REQUESTED** [review]

File: docs/dev/backport.md
Comment (avikivity) [reply docs/dev/backport.md:72]: I usually try the merge-commit first, and only if the changes are too complicated I fall back to individual commits.

File: docs/dev/backport.md
Comment (cvybhu) [reply docs/dev/backport.md:55]: Added a note that these instructions are for contributors, who were asked by maintainers to prepare a backport PR.

File: docs/dev/backport.md
Comment (cvybhu) [reply docs/dev/backport.md:68]: Damn git is terrible.

Added `git pull` before `submodule update`

File: docs/dev/backport.md
Comment (cvybhu) [reply docs/dev/backport.md:78]: Added a note that you can also use individual cherry-picks with the `-x` option.

File: docs/dev/backport.md
Comment (denesb) [reply docs/dev/backport.md:68]: In the usual workflow you already have `branch-5.0` and you just check it out, then do the sync and update and pull.

**Review by denesb — APPROVED** [review]

File: docs/dev/backport.md
Comment (cvybhu) [reply docs/dev/backport.md:68]: I'm no git expert, but my thinking was:
* use git fetch to fetch all commits from `branch-5.0` to the local repo
* use `git checkout -t origin/branch-5.0` to create a branch matching the remote one and checkout to it
* Now the local and version controlled submodule references could apparently be out of sync so I need to do `submodule sync`
* Then use `submodule update` to actually change the files in submodule directories to the right ones

I'm still not sure why the `git pull` is needed

**Comment by annastuchlik** [comment]
@cvybhu Thank you for this initiative to add the backport instructions - it's super useful!

File: docs/dev/backport.md
Comment (avikivity) [reply docs/dev/backport.md:68]: I run `git submodule foreach git fetch --all` to get commits for all submodules for all repos for which I have remotes.

**Pull request closed without merging**
